src/System.Net.HttpListener/tests/HttpRequestStreamTests.cs

```diff
@@ -0,0 +1,454 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Tests
```

Comment:
I agree - it's a good idea to port over the tests from Mono. I've taken a look at the Mono tests - not many duplicates with those added in this PR. The sheer number of Unix failures on this PR bolster this reasoning.